concent_api/utils/decorators.py

```diff
@@ -93,63 +93,70 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def handle_errors_and_responses(view):
+def handle_errors_and_responses(database_name):
     """
     Decorator for handling responses from Concent
-    for golem clients
+    for golem clients and handling transactions on given database name if it is not None.
     """
-    @wraps(view)
-    def wrapper(request, client_message, client_public_key, *args, **kwargs):
-        try:
-            sid = transaction.savepoint()
-            response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
-            transaction.savepoint_commit(sid)
-        except Http400 as exception:
-            logging.log_400_error(
-                view.__name__,
-                client_message,
-                client_public_key,
-            )
-            transaction.savepoint_rollback(sid)
-            return JsonResponse({'error': str(exception)}, status = 400)
-        if isinstance(response_from_view, message.Message):
-            assert response_from_view.sig is None
-            logging.log_message_returned(
-                response_from_view,
-                client_public_key,
-            )
-            serialized_message = dump(
-                response_from_view,
-                settings.CONCENT_PRIVATE_KEY,
-                client_public_key,
-            )
-
-            return HttpResponse(serialized_message, content_type = 'application/octet-stream')
-        elif isinstance(response_from_view, dict):
-            return JsonResponse(response_from_view, safe = False)
-        elif isinstance(response_from_view, HttpResponseNotAllowed):
-            logging.log_message_not_allowed(
-                view.__name__,
-                request.method,
-                client_public_key,
-            )
-            return response_from_view
-        elif isinstance(response_from_view, HttpResponse):
-            logging.log_message_accepted(
-                client_message,
-                client_public_key,
-            )
-            return response_from_view
-        elif response_from_view is None:
-            logging.log_empty_queue(
-                view.__name__,
-                client_public_key,
-            )
-            return HttpResponse("", status = 204)
-        elif isinstance(response_from_view, bytes):
-            return HttpResponse(response_from_view)
-
-        assert False, "Invalid response type"
-        raise Exception("Invalid response type")
+    assert database_name in settings.DATABASES
```

Comment:
Actually, due to this assert, `None` will not be accepted as `database_name`, thus if chescs below seem to be unnecessary. I reckon we should either enhance `settings.DATABASES` with `None` (does it make sense?) or get rid of checks.

Reply:
Yes, it does not handle `None`, but please do not assume that `None` will always be present in `settings.DATABASES`. It's not meant to be part of that dict.

`assert database_name in [settings.DATABASES, None]` would be just fine as long as there are special cases for `None` below.